### PR TITLE
Webpack TypeScript support

### PR DESCRIPTION
**Pull request by CitRUSprod**
Add script for start use TypeScript with Webpack.

### CODE REVIEW

File: scripts/setupTypeScriptWebpack.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+function addDepsToPackageJson() {
+	const pkgJSONPath = path.join(projectRoot, 'package.json');
+	const packageJSON = JSON.parse(fs.readFileSync(pkgJSONPath, 'utf8'));
+	packageJSON.devDependencies = Object.assign(packageJSON.devDependencies, {
+		'@tsconfig/svelte': '^1.0.10',
+		'@types/compression': '^1.7.0',
+		'@types/node': '^14.11.1',
+		'@types/polka': '^0.5.1',
+		'svelte-check': '^1.0.46',
+        'svelte-preprocess': '^4.3.0',
```
Comment (benmccann) [inline scripts/setupTypeScriptWebpack.js:63]: the spaces here should be tabs

**Comment by benmccann** [comment]
there's a lot of duplicated code between the webpack and rollup scripts. can we address that somehow?

**Comment by Conduitry** [comment]
I'd be fine with a single setup script that works for both and checks for the existence of `rollup.config.js` or `webpack.config.js` before deciding what to do.

Actually, it just struck me that we could refactor this all a bit and automatically produce four different branches (`rollup`, `rollup-typescript`, `webpack`, `webpack-typescript`) from a single branch, and get rid of the need for end users to deal with this script at all - but that should be a separate later PR.

**Comment by CitRUSprod** [comment]
> there's a lot of duplicated code between the webpack and rollup scripts. can we address that somehow?

Done.

**Review by benmccann — APPROVED** [review]

**Pull request merged**